### PR TITLE
feed issues #44

### PR DESCRIPTION
**Pull request by voila**
Fix atom feed validation errors

I used an urn:uuid: scheme for the feed id (provided on the command line via the --uuid flag) and entries id.

I've also added a new command line flag, --root, to provide the blog URL (some feed errors were due to URLs being relative)

### CODE REVIEW

File: config.ml [hunk 1]
```diff
@@ -53,6 +53,14 @@ let disk =
 
 (* Command-line options *)
 
+let root_k =
+  let doc = Key.Arg.info ~doc:"Blog URL" ["root"] in
+  Key.(create "root" Arg.(opt string "http://localhost" doc))
```
Comment (hannesm) [inline config.ml:58]: by default, canopy listens on port 8080 -- shouldn't this be reflected here in the url (and have the default value be `http://localhost:8080`).

**Comment by hannesm** [comment]
Thanks @voila -- this is imho a good intermediary solution for the feed issues.  But it is also clear that now the amount of command line arguments is slightly overwhelming, and we should move huge chunks into a configuration file stored in the data repository (using some key-value serialisation (yaml/sexp/json/younameit)).  This is not a reason to not merge this PR.

File: config.ml (hunk 1 above)
Comment (hannesm) [inline config.ml:58]: or maybe something where the default is built dynamically based on the values of port and tls?

**Comment by voila** [comment]
@hannesm 

> we should move huge chunks into a configuration file stored in the data repository (using some key-value serialisation (yaml/sexp/json/younameit)).

Should an issue be created for this ?

**Comment by hannesm** [comment]
On 26/06/2016 22:58, voila wrote:

> Should an issue be created for this ?

yes, sounds sensible to have an issue for that. there was somewhere in
this issue tracker some discussion about it...

**Comment by voila** [comment]
I later realised you had created #45

**Comment by hannesm** [comment]
@Engil good to get merged imho, what do you think?

**Comment by abbysmal** [comment]
Looks very nice, thank you for taking down this issue!
Adding two more command lines argument is ok for now, I will find some time in the next few weeks (most probably during the mirage hackathon) to migrate to a dedicated config file instead of relying on this many arguments…
One tiny detail is that, now we need the user to provide an uuid, which is probably the only solution to avoid generating many feeds troubles. Adding a tiny note somewhere in the doc about how to generate one easily would be nice I think. (something like a command line or a website to generate one would be alright I guess).
Thank you again for your work ! (and @hannesm for his careful reviews!)

**Comment by dbuenzli** [comment]
> Adding a tiny note somewhere in the doc about how to generate one easily would be nice I think. (something like a command line

If you have `uuidm` installed in opam you can do `uuidtrip -r` to generate a random uuid.

**Comment by voila** [comment]
I have added uuid generation to README and rebased onto master

**Comment by abbysmal** [comment]
Excellent!
Tested it earlier, everything looks alright, thank you again. :)

**Comment by hannesm** [comment]
thx, I've deployed this now on my canopy instance (needed new let's encrypt certificates anyways)

**Pull request merged**